tor/controller.go

```diff
@@ -373,23 +382,23 @@ func (c *Controller) readResponse(expected int) (int, string, error) {
 func parseTorReply(reply string) map[string]string {
 	params := make(map[string]string)
 
-	// Replies can either span single or multiple lines, so we'll default
-	// to stripping whitespace and newlines in order to retrieve the
-	// individual contents of it. The -1 indicates that we want this to span
-	// across all instances of a newline.
-	contents := strings.Split(strings.Replace(reply, "\n", " ", -1), " ")
+	// Find all fields of a reply. The -1 indicates that we want this to find all
+	// instances of the regexp.
+	contents := replyFieldRegexp.FindAllString(reply, -1)
 	for _, content := range contents {
 		// Each parameter within the reply should be of the form
-		// "KEY=VALUE". If the parameter doesn't contain "=", then we
-		// can assume it does not provide any other relevant information
-		// already known.
+		// KEY=VALUE or KEY="VALUE".
 		keyValue := strings.SplitN(content, "=", 2)
-		if len(keyValue) != 2 {
-			continue
-		}
-
 		key := keyValue[0]
 		value := keyValue[1]
+
+		// If it's a quoted value, remove quotes and backslashes before escaped
+		// characters. A backslash followed by any character represents that
+		// character.
+		if strings.HasPrefix(value, "\"") {
+			value = strings.ReplaceAll(value[1:len(value)-1], `\`, "")
```

Comment:
I think we shouldn't mix the back tick and the double quote notations here. Should we use back ticks for all three strings within this `if`?